src/Core/Commit.js

```diff
@@ -0,0 +1,314 @@
+/**
+ * @param {*} element
+ * @return {Element}
+ */
+function commitElement (element) {
+	if (element != null)
+		switch (element.constructor) {
+			case Element:
+				return element
+			case Array:
+				return elementFragment(element)
+			case String:
+			case Number:
+				return elementText(element)
+			case Function:
+			case Promise:
+				return createElement(element)
+			case Boolean:
+			case Symbol:
+				break
+			default:
+				return elementUnknown(element)
+		}
+
+	return elementText('')
+}
+
+/**
+ * @param {Element} element
+ * @param {Element} snapshot
+ */
+function commitPromise (element, snapshot) {
+	snapshot.type.then(function (value) {
+		if (!element.DOM)
+			return
+		if (element.flag === ElementPromise)
+			reconcileChildren(element, elementFragment(commitElement(value)))
+		else
+			reconcileElement(element, commitElement(value))
+	})
+}
+
+/**
+ * @param {Element} element
+ * @param {Element} sibling
+ * @param {Element} host
+ */
+function commitChildren (element, sibling, host) {
+	var children = element.children
+	var length = children.length
+	var next = children.next
+
+	while (length-- > 0) {
+		commitMount(!next.DOM ? next : merge(new Element(ElementNode), next), sibling, element, host, 0)
+		next = next.next
+	}
+}
+
+/**
+ * @param {Element} element
+ * @param {Element} sibling
+ * @param {Element} parent
+ * @param {Element} host
+ * @param {number} signature
+ */
+function commitMount (element, sibling, parent, host, signature) {
+	element.host = host
+	element.parent = parent
+	element.context = host.context
+
+ 	switch (element.flag) {
+ 		case ElementComponent:
+ 			element.work = WorkTask
+ 			
+ 			componentMount(element)
+
+ 			if (element.owner[LifecycleWillMount]) 
+ 				lifecycleMount(element, LifecycleWillMount)
+
+ 			commitMount(element.children, sibling, parent, element, signature)
+
+ 			if ((element.DOM = element.children.DOM, element.ref)) 
```

Comment:
this assigns then checks `element.ref`? Move the assignment outside of if expression?

Reply:
Yes, assigns element.DOM and checks element.ref.